### PR TITLE
[TwigComponent] Fix ux:icon & ux:map renders

### PR DESCRIPTION
**Pull request by smnandre**
Change implementation to avoid many dispatched events for nothing

Add a conflict in symfony/ux-map and symfony/ux-icons  to synchronise the 3 packages to 2.21

Performance gain around 20%

### CODE REVIEW

**Comment by Kocal** [comment]
> Performance gain around 20%

Can you link the Blackfire profiles delta plz?

**Review by javiereguiluz — APPROVED** [review]

**Review by kbond — APPROVED** [review]

**Comment by javiereguiluz** [comment]
Thank you Simon.

File: src/TwigComponent/src/DependencyInjection/TwigComponentExtension.php [hunk 1]
```diff
@@ -108,6 +110,10 @@ class_exists(AbstractArgument::class) ? new AbstractArgument(\sprintf('Added in
         $container->register('.ux.twig_component.twig.component_runtime', ComponentRuntime::class)
             ->setArguments([
                 new Reference('ux.twig_component.component_renderer'),
+                service_locator([
+                    'ux:icon' => service('.ux_icons.twig_icon_runtime')->nullOnInvalid(),
+                    'ux:map' => service('ux_map.twig_runtime')->nullOnInvalid(),
```
Comment (nicolas-grekas) [inline src/TwigComponent/src/DependencyInjection/TwigComponentExtension.php:115]: chiming here a bit late: this doesn't work
a service locator must point to non-null references
when these are turned into nulls, XmlDumper explodes (and that's a symptom of this code, which is not "legal")

Also, from a design pov, this looks super hardcoded. How can one pass more things to the locator?

This should use a tagged locator instead.

Comment (nicolas-grekas) [reply src/TwigComponent/src/DependencyInjection/TwigComponentExtension.php:115]: This was spotted by the CI BTW:
https://github.com/symfony/ux/actions/runs/11071829929/job/30764609180

**Pull request merged**